zkevm-circuits/src/keccak_circuit/test.rs

```diff
@@ -16,17 +26,82 @@ fn verify<F: Field>(k: u32, inputs: Vec<Vec<u8>>, success: bool) {
         }
         panic!();
     }
+
+    // Extract the content of the lookup table.
+    let hashes = {
+        // Find the columns of the table.
+        let is_enabled = prover.advice_values(config.keccak_table.is_enabled);
+        let input_rlc = prover.advice_values(config.keccak_table.input_rlc);
+        let input_len = prover.advice_values(config.keccak_table.input_len);
+        let output_rlc = prover.advice_values(config.keccak_table.output_rlc);
+
+        // Keep the rows that are supposed to contain hash results.
+        izip!(is_enabled, input_rlc, input_len, output_rlc)
+            .filter(|(enabled, _, _, _)| assigned_non_zero(enabled))
+            .map(|(_, input_rlc, input_len, output_rlc)| {
+                (unwrap(input_rlc), unwrap(input_len), unwrap(output_rlc))
+            })
+            .collect::<Vec<(F, F, F)>>()
+    };
+
+    // Calculate the expected inputs in reversed-RLC form.
+    let rlc_input = |bytes: &[u8]| rlc::value(bytes.iter().rev(), input_challenge);
+
+    // Calculate the expected digests in reversed-RLC form.
+    let rlc_digest = |hex_str: &str| {
+        let bytes = hex::decode(hex_str).unwrap();
+        rlc::value(bytes.iter().rev(), hash_challenge)
+    };
+
+    // Check that all the digests are there.
+    assert!(hashes.len() >= inputs.len());
+    assert_eq!(inputs.len(), digests.len());
+    for (input, digest, hash) in izip!(&inputs, &digests, &hashes) {
+        let len = F::from(input.len() as u64);
+        let expected = (rlc_input(input), len, rlc_digest(digest));
+        assert_eq!(*hash, expected);
+    }
+
+    // Check that other digests are the digest of the empty message.
+    assert_eq!(F::zero(), rlc_input(&inputs[0]));
+    let empty_hash = (F::zero(), F::zero(), rlc_digest(&digests[0]));
+    for hash in hashes.iter().skip(inputs.len()) {
+        assert_eq!(*hash, empty_hash);
+    }
```

Comment:
it is a bit counter-intuitive that we make assumption on the tests data... I think just replacing inputs[0] with "vec![]", and replacing digests[0] with c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470 will be good.

Reply:
Makes sense, done.